system/jlib/jtrace.cpp

```diff
@@ -161,9 +161,12 @@ class CSpan : public CInterfaceOf<ISpan>
 
     virtual void beforeDispose() override
     {
-        StringBuffer out;
-        toLog(out);
-        LOG(MCmonitorEvent, "Span end: {%s}", out.str());
+        if (queryTraceManager().logSpanFinish())
```

Comment:
If we decide to only log once per span, I wonder if we could filter all MCmonitorEvent directly at the jlog level.

Reply:
We could, but it is more efficient to not generate it at all.